libyul/backends/wasm/WasmCodeTransform.cpp

```diff
@@ -346,13 +361,29 @@ string WasmCodeTransform::newLabel()
 	return m_nameDispenser.newName("label_"_yulstring).str();
 }
 
-void WasmCodeTransform::allocateGlobals(size_t _amount)
+vector<size_t> WasmCodeTransform::allocateGlobals(vector<wasm::Type> const& _typesForGlobals)
 {
-	while (m_globalVariables.size() < _amount)
-		m_globalVariables.emplace_back(wasm::GlobalVariableDeclaration{
-			m_nameDispenser.newName("global_"_yulstring).str(),
-			wasm::Type::i64
-		});
+	vector<size_t> allocatedIndices;
+
+	map<wasm::Type, size_t> firstUnused;
+	for (wasm::Type const& type: _typesForGlobals)
+	{
+		firstUnused.try_emplace(type, 0);
```

Comment:
The next line does this implicitly, doesn't it?

Reply:
It does. I can remove it but I felt that initializing the counter explicitly with zero would be better for readability.

I could also do this:
``` c++
map<wasm::Type, size_t> firstUnused = {{wasm::Type::i32, 0}, {wasm::Type::i64, 0}};
```
but it would require me to list all the types explcitly and I liked the fact that in the current form the function will not have to change at all if we ever add more types (e.g. `f32`/`f64`).

Should I remove it?

Reply:
No, it's fine!